kv/dist_sender.go

```diff
@@ -396,6 +403,15 @@ func (ds *DistSender) Send(call *client.Call) {
 				}
 			} else if isMulti {
 				// If this request spans ranges, collect the replies.
+				if maxResults > 0 {
+					if reply, ok := reply.(proto.Truncatable); ok {
```

Comment:
Thanks for your comments, @tschottdorf , one small suggestions: If we put both maxResults and accumulateResults into the "isMulti" branch, then each time when accessing one range inside a multi-range request, accumulatedResults needs to be calculated again according to the received responses, some performance burden, right? maybe it is better to keep accumulatedResults outside.